src/app.md

```diff
@@ -271,6 +272,40 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-file-skip-dot]: #ex-file-skip-dot
+<a name="ex-file-skip-dot"></a>
+##  Traverse directories while skipping dotfiles
+
+Uses [`WalkDirIterator::filter_entry`] to descend recursively into entries passing the `is_not_hidded` predicate thus skipping hidden files and directories whereas [`Iterator::filter`] would be applied to each `DirEntry` even if the parent is a hidden directory. 
+
+In the predicate [`WalkDir::depth`] skips level 0 which is the root directory `"."`.
```

Comment:
While its true that the predicate itself "skips level 0". I feel that it breaks readers a mental flow (requiring them inspect the code in order to understand why we can actually avoid skipping the root which is the main idea).

Mostly we aim for the example pros to be self explanatory so I
I would prefer this to mention that "Root dir is never skipped due to [`WalkDir::depth`] usage in `is_not_hidded` predicate".

Then readers know what is going on without back and forth to the actual code.